### PR TITLE
RFC: examples: make conditional branching example a standalone module

### PR DESCRIPTION
**Pull request by yuhan**
## Summary
why this change?
I plan to clean up https://github.com/dagster-io/dagster/tree/master/examples and make the structure like:
```
examples
  - docs_snippets
  
  # how to deploy dagster to ...
  - deploy-docker
  - deploy-ecs
  - deploy-k8s
  
  # how to use dagster with ...
  - with-great-expectations
  - with-dbt
  - with-pyspark
  - with-pyspark-emr
  
  # how to use ...
  - dagster-conditional-branching
  - dagster-dynamic-out
  - dagster-memoized-development
  - ... and the other dagster features
  
  # larger demos/examples
  - hacker_news
  - hacker_news_assets
```
then, make each folder standalone that users can clone, run jobs, load dagit, and deploy - WIP: https://github.com/dagster-io/dagster/pull/5805

why hyphen?
1) it makes it clear that etl-conditional-branching is just the directory that includes 
2) low impact but still: presumably we'd want to include gh links to example folders from docs site or dagster.io. hyphens in links are better for SEO because Google treats a hyphen as a word separator, but doesn't treat underscore the same way.

## Checklist

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by yuhan** [comment]
Current dependencies on/for this PR:
* master
  * **PR #5731** 
    * **PR #5787**  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh) and will continue to be automatically updated while this PR remains open.

**Comment by sryza** [comment]
If someone asks "how do I do conditional branching in Dagster?", would we want to send them to this example or to the [conditional branching example on the docs concepts page](https://docs.dagster.io/concepts/ops-jobs-graphs/jobs-graphs#conditional-branching)?

**Review by yuhan — COMMENTED** [review]
I'm imagining that we would send them to either based on what they are asking for:
- if they know what they are doing and they just want to see how to use the api, we'd send them to [the one of the concept page](https://docs.dagster.io/concepts/ops-jobs-graphs/jobs-graphs#conditional-branching) for "api usage"
- if they bring their use case and ask for recommendations (which has been most of questions that we ended up answering with conditional branching), we'd send them to this example for "here's an example for accomplishing a similar use case + how you can connect dots among multiple dagster concepts". 

in other words, i view the current examples on concept pages as api usage while this example would focus more on the "why"s and using 2+ dagster concepts (i.e. more of a dagster project that can actually launch runs, as opposed to small code snippets).

**Review by dpeng817 — APPROVED** [review]
All on board once this passes tests.Probably some tox related stuff.

**Comment by sryza** [comment]
Got it - the difference between showcasing APIs and showing how to implement use cases makes sense to me.

A couple concerns though:
* In the case of this conditional branching example, I wasn't able to tell just from reading it what use case it's trying to implement and why conditional branching is relevant to that use case.  I.e. why do we want to write the "all" csv, but not the "nyc" csv?  Wouldn't that mean that, if yesterday's homepage had "nyc" articles, but today's doesn't, the "nyc" csv would end up with data that's out of date and out of sync with the "all" csv? It would be helpful to include some text that describes what's going on, and to give the job a name that's representative of the use case, rather than one of the concepts it's showcasing.
* As you mentioned, examples can showcase multiple concepts.  Why call this one "conditional-branching" instead of "custom-io-manager"? Would it make sense to call it something like "fetch-nyt-articles" and to have the README include tags for all the concepts it uses.

**Comment by yuhan** [comment]
@sryza thanks for the feedback! The concerns are all very reasonable and here are my thoughts:
- Will update the processing logic and add more description in a later stack. The idea of this example is we take all the data, parse the data into category info and store that info as "all" csv, and while we are processing the entire data we also wanna do something special with the nyc articles (e.g. store all the nyc articles and send a slack alert so i can read it), which is where conditional branching becomes relevant. If today's data doesn't have nyc articles, that branch will be skipped - this is why we want conditional branching (aka some branching logic based on run time value)
- We call this conditional-branching because we are planning to write more small ones that primarily focus on the shape of the graph. Here's another wip example: https://github.com/dagster-io/dagster/pull/5822. I'm not attached to the exact name, but my plan is to have a few more out with ok names and ok project structures first, and then once we have, say, 3 or 4 examples (e.g. conditional-branching, dynamic-out, with s3 and redshift), we should have a better sense for the following and iterate:
  1) better project structures for examples like these - which could be an input to the "recommended project structure".
  2) better naming scheme. 
  
In terms of the plan and next steps, i'll be more full time thinking about this and more clear about the plan once im done with the op selection work (feel bad that this workstream has been a bit slow and up in the air these days). Here are some concrete next steps: 
  1) improve the conditional branching example to use io manager and have better description and op names.
  2) finishing up https://github.com/dagster-io/dagster/pull/5822
  3) start another one for aws etl (focusing on etl with s3/redshift resources/io manager and jobs for multiple envs).
